pkg/analytics/job_submit.go

```diff
@@ -0,0 +1,91 @@
+package analytics
+
+import (
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+// SubmitJobEventType is the event type for a job that has been submitted to an orchestrator.
+const SubmitJobEventType = "bacalhau.job_v1.submit"
+
+type SubmitJobEvent struct {
+	JobID         string `json:"job_id"`
+	NameSet       bool   `json:"name_set"`
+	NamespaceHash string `json:"namespace_hash"`
+	Type          string `json:"type"`
+	Count         int    `json:"count"`
+	LabelsCount   int    `json:"labels_count"`
+	MetaCount     int    `json:"meta_count"`
+	Version       uint64 `json:"version"`
+	Revision      uint64 `json:"revision"`
+	CreateTime    int64  `json:"create_time"`
+	ModifyTime    int64  `json:"modify_time"`
+
+	TaskNameHash         string   `json:"task_name_hash"`
+	TaskEngineType       string   `json:"task_engine_type"`
+	TaskPublisherType    string   `json:"task_publisher_type"`
+	TaskEnvVarCount      int      `json:"task_env_var_count"`
+	TaskMetaCount        int      `json:"task_meta_count"`
+	TaskInputSourceTypes []string `json:"task_input_source_types"`
+	TaskResultPathCount  int      `json:"task_result_path_count"`
+	TaskCPUUnits         float64  `json:"task_cpu_units"`
+	TaskMemoryBytes      uint64   `json:"task_memory_bytes"`
+	TaskDiskBytes        uint64   `json:"task_disk_bytes"`
+	TaskGPUCount         uint64   `json:"task_gpu_count"`
+	TaskNetworkType      string   `json:"task_network_type"`
+	TaskDomainsCount     int      `json:"task_domains_count"`
+	TaskExecutionTimeout int64    `json:"task_execution_timeout"`
+	TaskQueueTimeout     int64    `json:"task_queue_timeout"`
+	TaskTotalTimeout     int64    `json:"task_total_timeout"`
+
+	Warnings []string `json:"warnings"`
+	Error    string   `json:"error"`
+}
+
+func NewSubmitJobEvent(j models.Job, warnings ...string) SubmitJobEvent {
+	t := j.Task()
+	taskInputTypes := make([]string, len(t.InputSources))
+	for i, s := range t.InputSources {
+		taskInputTypes[i] = s.Source.Type
+	}
+	// if we can't parse the resources use zero
+	taskResources, err := t.ResourcesConfig.ToResources()
+	if err != nil {
+		taskResources = &models.Resources{
+			CPU:    0,
+			Memory: 0,
+			Disk:   0,
+			GPU:    0,
+			GPUs:   nil,
```

Comment:
we need to capture types of GPUs used as well

Reply:
This information isn't included in a task.